src/vs/editor/contrib/codeAction/browser/codeActionCommands.ts

```diff
@@ -156,14 +156,21 @@ export class QuickFixController extends Disposable implements IEditorContributio
 	}
 
 	private _applyCodeAction(action: CodeActionItem, preview: boolean): Promise<void> {
-		return this._instantiationService.invokeFunction(applyCodeAction, action, { preview, editor: this._editor });
+		return this._instantiationService.invokeFunction(applyCodeAction, action, { preview, editor: this._editor }, applyCodeActionReason.FromCodeActions);
 	}
 }
 
+export enum applyCodeActionReason {
+	OnSave = 'onSave',
+	FromProblemsView = 'fromProblemsView',
+	FromCodeActions = 'fromCodeActions'
+}
+
 export async function applyCodeAction(
 	accessor: ServicesAccessor,
 	item: CodeActionItem,
-	options?: { preview?: boolean; editor?: ICodeEditor }
+	options?: { preview?: boolean; editor?: ICodeEditor },
+	codeActionReason?: applyCodeActionReason,
```

Comment:
If anyone starts calling this function, it will be easy for them to forget to pass in a `codeActionReason`. Can you make this parameter non-optional (you'll have to either change the order of the parameters or make `options` non-optional too for this)